### ISSUE TITLE
Unlinking a tempfile?

### ISSUE THREAD

**Issue by jcoyne** [projecthydra/hydra-derivatives#77]
```
     No such file or directory @ unlink_internal - /tmp/20516b3c-9efa-4b2e-b471-33fbdd822da7-120150827-22031-3paka9.pdf

    /opt/goldenseal/shared/bundle/ruby/2.2.0/gems/hydra-derivatives-2.0.0/lib/hydra/derivatives/document.rb:20:in `unlink'
    /opt/goldenseal/shared/bundle/ruby/2.2.0/gems/hydra-derivatives-2.0.0/lib/hydra/derivatives/document.rb:20:in `block in encode_file'
    /opt/goldenseal/shared/bundle/ruby/2.2.0/gems/hydra-derivatives-2.0.0/lib/hydra/derivatives/services/tempfile_service.rb:34:in `block in default_tempfile'
    /usr/local/lib/ruby/2.2.0/tempfile.rb:319:in `open'
    /opt/goldenseal/shared/bundle/ruby/2.2.0/gems/hydra-derivatives-2.0.0/lib/hydra/derivatives/services/tempfile_service.rb:25:in `default_tempfile'
    /opt/goldenseal/shared/bundle/ruby/2.2.0/gems/hydra-derivatives-2.0.0/lib/hydra/derivatives/services/tempfile_service.rb:20:in `tempfile'
    /opt/goldenseal/shared/bundle/ruby/2.2.0/gems/hydra-derivatives-2.0.0/lib/hydra/derivatives/services/tempfile_service.rb:7:in `create'
    /opt/goldenseal/shared/bundle/ruby/2.2.0/gems/hydra-derivatives-2.0.0/lib/hydra/derivatives/document.rb:14:in `encode_file'
    /opt/goldenseal/shared/bundle/ruby/2.2.0/gems/hydra-derivatives-2.0.0/lib/hydra/derivatives/shell_based_processor.rb:22:in `block in process'
    /opt/goldenseal/shared/bundle/ruby/2.2.0/gems/hydra-derivatives-2.0.0/lib/hydra/derivatives/shell_based_processor.rb:17:in `each'
    /opt/goldenseal/shared/bundle/ruby/2.2.0/gems/hydra-derivatives-2.0.0/lib/hydra/derivatives/shell_based_processor.rb:17:in `process'
    /opt/goldenseal/shared/bundle/ruby/2.2.0/gems/hydra-derivatives-2.0.0/lib/hydra/derivatives.rb:104:in `transform_file'
    /opt/goldenseal/shared/bundle/ruby/2.2.0/bundler/gems/hydra-works-dc0bfd08eb24/lib/hydra/works/models/concerns/generic_file/derivatives.rb:17:in `block (2 levels) in <module:Derivatives>'
    /opt/goldenseal/shared/bundle/ruby/2.2.0/gems/hydra-derivatives-2.0.0/lib/hydra/derivatives.rb:66:in `call'
    /opt/goldenseal/shared/bundle/ruby/2.2.0/gems/hydra-derivatives-2.0.0/lib/hydra/derivatives.rb:66:in `block in create_derivatives'
    /opt/goldenseal/shared/bundle/ruby/2.2.0/gems/hydra-derivatives-2.0.0/lib/hydra/derivatives.rb:64:in `each'
    /opt/goldenseal/shared/bundle/ruby/2.2.0/gems/hydra-derivatives-2.0.0/lib/hydra/derivatives.rb:64:in `create_derivatives'
    /opt/goldenseal/shared/bundle/ruby/2.2.0/bundler/gems/curation_concerns-9b49d4d66ed5/curation_concerns-models/app/jobs/create_derivatives_job.rb:10:in `run'
    /opt/goldenseal/shared/bundle/ruby/2.2.0/bundler/gems/curation_concerns-9b49d4d66ed5/curation_concerns-models/lib/curation_concerns/models/resque.rb:32:in `perform'


```

**Comment by tampakis** [comment]
I've come across this problem before. I think it's because you have LibreOffice/whatever program is used to create document derivatives open on your machine.

**Comment by jcoyne** [comment]
@tampakis not likely, this is on a headless server.

**Comment by jcoyne** [comment]
Possibly fixed by 29b32e2c93734e8535c93d5ba0cec184ef2fb18e
